server/src/main/java/com/paypal/selion/proxy/SeLionSauceProxy.java

```diff
@@ -97,14 +273,59 @@ public URL getRemoteHost() {
         return remoteHost;
     }
 
-    public void teardown() {
-        super.teardown();
+    @Override
+    public void beforeSession(TestSession session) {
+        // nothing to do before session
     }
 
-    public boolean isAlive() {
-        // If the sauce server is down the grid is marking node as down. To avoid this it is overridden to always return
-        // true.
-        return true;
+    @Override
+    public void afterSession(TestSession session) {
+        // nothing to do after session
+    }
+
+    @Override
+    public void startPolling() {
+        // polls but does not disconnect the node, once down polling limit is reached
+        pollingThread = new Thread(new Runnable() { // Thread safety reviewed
+            int failedPollingTries; // 0 is the default
+            long downSince; // 0L is the default
+
+            public void run() {
+                while (poll) {
+                    try {
+                        Thread.sleep(pollingInterval);
+                        if (!isAlive()) {
+                            if (!down) {
+                                failedPollingTries++;
+                                if (failedPollingTries >= downPollingLimit) {
+                                    downSince = System.currentTimeMillis();
+                                    addNewEvent(new RemoteNotReachableException(String.format(
```

Comment:
this is broken.. it won't change the status of `down` to `true` for `SeLionSauceProxy`